test/Libraries/CoreNodeModelsTests/SelectionTests.cs

```diff
@@ -0,0 +1,188 @@
+            var startingAmount = selection.SelectionResults.Count();
+            Assert.AreEqual(0, startingAmount);
+
+            //Selects the test node
+            selectionHelperMock
+                .Setup(x => x.RequestSelectionOfType("testMessage", SelectionType.Many, SelectionObjectType.Element))
+                .Returns(new List<ModelBase> { testNode });
+            selection.Select(testNode);
+
+            //checks one node is selected
+            var result = selection.SelectionResults.Count();
+            Assert.AreEqual(1, result);
+
+            Mock.VerifyAll();
+        }
+
+        [Test]
+        public void ClearSelectionTest()
+        {
+            //Selects the test node
+            selectionHelperMock
+                .Setup(x => x.RequestSelectionOfType("testMessage", SelectionType.Many, SelectionObjectType.Element))
+                .Returns(new List<ModelBase> { testNode });
+            selection.Select(testNode);
+
+            //checks one node is selected
+            var cant = selection.SelectionResults.Count();
+            Assert.AreEqual(1, cant);
+
+            selection.ClearSelections();
+            //Checks no nodes are selected
+            var result = selection.SelectionResults.Count();
+            Assert.AreEqual(0, result);
+
+            Mock.VerifyAll();
+        }
+
+        [Test]
+        public void ToStringTest()
+        {
+            //Checks ToString result for an empty selection
+            var toStringResult = selection.ToString();
+            Assert.AreEqual(Resources.SelectionNodeNothingSelected, toStringResult);
+
+            //Selects the test node
+            selectionHelperMock
+                .Setup(x => x.RequestSelectionOfType("testMessage", SelectionType.Many, SelectionObjectType.Element))
+                .Returns(new List<ModelBase> { testNode });
+            selection.Select(testNode);
+
+            //Checks ToString result for the selected element
+            toStringResult = selection.ToString();
+            Assert.AreEqual("testPrefix : Dynamo.Graph.Nodes.CodeBlockNodeModel", toStringResult);
+
+            Mock.VerifyAll();
+        }
+
+        /// <summary>
+        /// Class created in order to test protected methods in SelectionBase parent
+        /// </summary>
+        private class SelectionConcrete : SelectionBase<ModelBase, ModelBase>
+        {
+            public override IModelSelectionHelper<ModelBase> SelectionHelper { get; }
+
+            public void SerializeCore(XmlElement nodeElement, SaveContext context) =>
+                SerializeCore(nodeElement, context);
+
+            public void DeserializeCore(XmlElement nodeElement, SaveContext context) =>
+                DeserializeCore(nodeElement, context);
+
+            public bool UpdateValueCore(UpdateValueParams updateValueParams) =>
+                UpdateValueCore(updateValueParams);
+
+            public string GetOutputPortName() =>
+                 GetOutputPortName();
```

Comment:
Why implement these when they're simply calling the base methods?

Reply:
the base methods are protected, this way I can call them for testing

Reply:
also, you may notice some of this arent being tested yet, those tests are the ones im working on right now and will be added on a future commit soon